azure/Guidelines.md

```diff
@@ -0,0 +1,864 @@
+ 
+> :warning: **YOU SHOULD NOT** adopt this pattern. If not following this advice, you MUST disclose your reason during the Azure HTTP/REST Stewardship Board review.
+ 
+> :no_entry: **DO NOT** adopt this pattern. If you feel you need an exception, contact the Azure HTTP/REST Stewardship Board <b>prior</b> to implementation.
+ 
+*If you feel you need an exception, or need clarity based on your situation, please contact the Azure HTTP/REST Stewardship Board <b>prior</b> to release of your API.*
+
+## Building Blocks: HTTP, REST, & JSON
+The Microsoft Azure Cloud platform exposes its APIs through the core building blocks of the Internet; namely HTTP, REST, and JSON. This section provides you with a general understanding of how these technologies should be applied when creating your service.
+
+### HTTP
+Azure services must adhere to the HTTP specification, [RFC7231](https://tools.ietf.org/html/rfc7231). This section further refines and constrains how service implementors should apply the constructs defined in the HTTP specification. It is therefore, important that you have a firm understanding of the following concepts:
+* [Uniform Resource Locators (URLs)](URLS)
+* HTTP Methods
+* Request & Response Headers
+* Bodies
+
+### Uniform Resource Locators (URLs)
+
+A Uniform Resource Locator (URL) is how developers access the resources of your service. Ultimately, URLs are how developers form a cognitive model of your service's resources.
+
+> :white_check_mark: **DO** use this URL pattern:
+```text
+https://<service>.<cloud>/<tenant>/<service-root>/<resource-collection>/<resource-id>/
+```
+
+Where:
+ | Field | Description
+ | - | - |
+ | service | Name of the service (ex: blobstore, servicebus, directory, or management)
+ | cloud | Cloud domain name (see Azure CLI's "az cloud list")<p><table><tr><td><b>Cloud</td><td><b>Domain</td></tr><tr><td>Public</td><td>azure.net</td></tr><tr><td>US Government</td><td>usgovcloudapi.net</td></tr><tr><td>China</td><td>chinacloudapi.cn</td></tr><tr><td>German</td><td>cloudapi.de</td></tr></table>
+ | tenant | Globally-unique ID of container representing tenant isolation, billing, enforced quotas, lifetime of containees (ex: subscription UUID)
+ | service&#x2011;root | Service-specific path (ex: blobcontainer, myqueue)
+ | resource&#x2011;collection | Name of the collection, unabbreviated, pluralized
+ | resource&#x2011;id | Value of the unique id property. This MUST be the raw string/number/guid value with no quoting but properly escaped to fit in a URL segment.
+
+> :white_check_mark: **DO** use kebab-casing (preferred) or camel-casing for URL path segments. If the segment refers to a JSON field, use camel casing.
+> 
+> :white_check_mark: **DO** return ```414-URI Too Long``` if a URL exceeds 2083 characters
+>
+> :white_check_mark: **DO** treat URL path segments as case-sensitive. If the passed-in case doesn't match what the service expects, the request __MUST__ fail with the appropriate HTTP return code.
```

Comment:
What are the appropriate codes for this condition?

Reply:
```suggestion
> :white_check_mark: **DO** treat URL path segments as case-sensitive. If the passed-in case doesn't match what the service expects, the request __MUST__ fail with the HTTP ```404 Not Found``` return code.
```